test/apps/iodemo/io_demo.cc

```diff
@@ -1612,18 +1618,15 @@ class DemoClient : public P2pDemoCommon {
                     << msg->sn << " data size " << msg->data_size
                     << " conn " << conn;
 
+        assert(conn->ucx_status() == UCS_OK);
+
+        size_t server_index = get_active_server_index(conn);
+        assert(server_index < _server_info.size());
```

Comment:
maybe no need to retrieve server_index if it's not a completion? just for reducing overheads

Reply:
good idea, done